lib/active_model/serializer/associations.rb

```diff
@@ -73,12 +76,19 @@ def has_one(name, options = {})
         #
         # @api private
         #
-        def associate(reflection)
+        def associate(reflection, block)
           self._reflections = _reflections.dup
 
-          define_method reflection.name do
-            object.send reflection.name
-          end unless method_defined?(reflection.name)
+          reflection_name = reflection.name
+          if block
+            serialized_associations[reflection_name] = ->(instance) { instance.instance_eval(&block) }
+          else
+            serialized_associations[reflection_name] = ->(instance) { instance.object.send(reflection_name) }
+          end
+
+          define_method reflection_name do
+            serialized_associations[reflection_name].call(self)
+          end unless method_defined?(reflection_name)
```

Comment:
TODO: [ ] remove dynamic method definition from associations.  (I sorta wish this weren't a module mixed into Serializer, but oh well)